tests/unit/dojson/test_dojson_hepnames.py

```diff
@@ -55,16 +58,195 @@ def test_dates(marcxml_to_json, json_to_marc):
+            <subfield code="d">2020</subfield>
+            <subfield code="e">CERN-ALPHA</subfield>
+            <subfield code="0">00001</subfield>
+            <subfield code="s">2014</subfield>
+            <subfield code="z">current</subfield>
+        </datafield>
+        <datafield tag="693" ind1=" " ind2=" ">
+            <subfield code="d">2012</subfield>
+            <subfield code="e">CERN-ALPHA</subfield>
+            <subfield code="0">00001</subfield>
+            <subfield code="s">2010</subfield>
+        </datafield>
+        ''',
+        [
+            {
+                'curated_relation': True,
+                'current': True,
+                'end_year': 2020,
+                'name': 'CERN-ALPHA',
+                'record': in_context_get_record_ref('00001'),
+                'start_year': 2014,
+            },
+            {
+                'curated_relation': True,
+                'current': False,
+                'end_year': 2012,
+                'name': 'CERN-ALPHA',
+                'record': in_context_get_record_ref('00001'),
+                'start_year': 2010,
+            },
+        ],
+        [
+            {
+                '0': 1,
+                'd': 2020,
+                'e': 'CERN-ALPHA',
+                's': 2014,
+                'z': 'current',
+            },
+            {
+                '0': 1,
+                'd': 2012,
+                'e': 'CERN-ALPHA',
+                's': 2010,
+            },
+        ],
+    ),
+    (
+        'simultaneous_experiments',
+        '''
+        <datafield tag="693" ind1=" " ind2=" ">
+            <subfield code="d">2013</subfield>
+            <subfield code="e">FIRST-SIMULTANEOUS</subfield>
+            <subfield code="e">SECOND-SIMULTANEOUS</subfield>
+            <subfield code="0">00001</subfield>
+            <subfield code="0">00002</subfield>
+            <subfield code="s">2015</subfield>
+        </datafield>
+        ''',
+        [
+            {
+                'curated_relation': True,
+                'current': False,
+                'end_year': 2013,
+                'name': 'FIRST-SIMULTANEOUS',
+                'record': in_context_get_record_ref('00001'),
+                'start_year': 2015,
+            },
+            {
+                'curated_relation': True,
+                'current': False,
+                'end_year': 2013,
+                'name': 'SECOND-SIMULTANEOUS',
+                'record': in_context_get_record_ref('00002'),
+                'start_year': 2015
+            },
+        ],
+        [
+            {
+                '0': 1,
+                'd': 2013,
+                'e': 'FIRST-SIMULTANEOUS',
+                's': 2015,
+            },
+            {
+                '0': 2,
+                'd': 2013,
+                'e': 'SECOND-SIMULTANEOUS',
+                's': 2015,
+            },
+        ],
+    ),
+
+]
+
+
+@pytest.mark.parametrize("data", EXPERIMENTS_DATA, ids=idfn)
```

Comment:
```
_____________________________________ test_experiments[simultaneous_experiments] ______________________________________

experiments_data = ('simultaneous_experiments', '\n            <datafield tag="693" ind1=" " ind2=" ">\n                <subfield code="d...{'0': 1, 'd': 2013, 'e': 'FIRST-SIMULTANEOUS', 's': 2015}, {'0': 2, 'd': 2013, 'e': 'SECOND-SIMULTANEOUS', 's': 2015}])

    def test_experiments(experiments_data):
        """Test if experiments is created correctly."""
        xml_snippet, expected_json, expected_marc = experiments_data[1:]
        if not xml_snippet.strip().startswith('<record>'):
            xml_snippet = '<record>%s</record>' % xml_snippet
    
        json_data = clean_record(hepnames.do(create_record(xml_snippet)))
        json_experiments = json_data['experiments']
        marc_experiments = hepnames2marc.do(json_data)['693']
    
        assert marc_experiments == expected_marc
>       assert json_experiments == expected_json
E       assert [{'curated_re...ANEOUS', ...}] == [{'curated_rel...ANEOUS', ...}]
E         At index 0 diff: {'start_year': 2015, 'name': 'FIRST-SIMULTANEOUS', 'record': {'$ref': 'http://localhost:5000/api/record/00001'}, 'end_year': 2013, 'current': False, 'curated_relation': True} != {'start_year': 2015, 'name': 'FIRST-SIMULTANEOUS', 'record': {'$ref': 'http://inspirehep.net/api/record/00001'}, 'end_year': 2013, 'current': False, 'curated_relation': True}
E         Use -v to get the full diff

tests/unit/dojson/test_dojson_hepnames.py:246: AssertionError
======================================== 4 failed, 36 passed in 11.43 seconds =========================================
```

there you go, it's more readable if you run with verbose, and even easier to debug if you run with --pdb (or --ipdb if ipdb is installed in the venv, or --bpdb), the original xml, the original parsed xml, the json result and the expectation and the parsed marc result and expectation are locally in that env too, so just a print of the one you want to check will show the contents.

Still trying to figure out how to use the 'app' fixture without the modules... does not seem an easy task... it looks very strange that pytest does not recommend reusing the code as modules...